parlai/agents/transformer/modules.py

```diff
@@ -356,6 +358,13 @@ def forward(self, input, positions=None, segments=None):
         if self.embeddings_scale:
             tensor = tensor * np.sqrt(self.dim)
 
+        if positions.shape[1] > self.n_positions:
+            warn_once(
+                'You are inputting a sequence of {x} length, but only have'
+                '--n-positions {y}. Set --truncate or increase --n-positions'.format(
+                    x=positions.shape[1],
```

Comment:
I told you `positions.shape[1]`, but let's do `positions.max().item()` instead. You can imagine a situation where positions is resetting.

Do you mind also adding in the check for segments too?